### PR TITLE
Update Convolution-Layers.md

### PR DESCRIPTION
**Pull request by helenlly**
## What changes were proposed in this pull request?

correct wrong content about LocallyConnected2D:
1) change LocallyConnected1D -_ LocallyConnected2D
2) modify scala param as  https://github.com/intel-analytics/BigDL/blob/master/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected2D.scala
3) modify python param as https://github.com/intel-analytics/BigDL/blob/master/pyspark/bigdl/nn/layer.py#L1310
4)add a link to  SpatialConvolution

## How was this patch tested?

tested in local

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2191

### CODE REVIEW

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 1]
```diff
@@ -1831,49 +1831,49 @@ output m is : [[[[[ 0.80314147  0.80314147  0.7915861   0.7915861 ]
...
+   nInputPlane, inputWidth, inputHeight, nOutputPlane, kernelW, kernelH, 
+   strideW = 1, strideH = 1, padW = 0, padH = 0, propagateBack = true, 
+   wRegularizer = null, bRegularizer = null, initWeight = null, initBias = null,
+   initGradWeight = null, initGradBias = null, withBias = true, format = DataFormat.NCHW
+  )
+```
+**Python:**
+```python
+module = LocallyConnected2D(
+    self, n_input_plane, input_width, input_height, n_output_plane,
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: need to remove `self,`

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 2]
```diff
@@ -1831,49 +1831,49 @@ output m is : [[[[[ 0.80314147  0.80314147  0.7915861   0.7915861 ]
...
+    init_grad_bias=None, with_bias=True, data_format="NCHW", bigdl_type="float"
+  )
+```                            
+
+The LocallyConnected2D layer works similarly to the [SpatialConvolution](APIGuide/Layers/Convolution-Layers.md#SpatialConvolution) layer, except that weights are unshared, that is, a different set of filters is applied at each different patch of the input.
 
     * `n_input_plane` The number of expected input planes in the image given into forward()
-    * `input_width` The expected width of input
-    * `input_height` The expected height of input
+    * `input_width` The input width
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: inputWidth , inputHeight

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 3]
```diff
@@ -1831,49 +1831,49 @@ output m is : [[[[[ 0.80314147  0.80314147  0.7915861   0.7915861 ]
...
+    kernel_w, kernel_h, stride_w=1, stride_h=1, pad_w=0, pad_h=0，
+    propagate_back=True wRegularizer=None, bRegularizer=None, 
+    init_weight=None, init_bias=None, init_grad_weight=None, 
+    init_grad_bias=None, with_bias=True, data_format="NCHW", bigdl_type="float"
+  )
+```                            
+
+The LocallyConnected2D layer works similarly to the [SpatialConvolution](APIGuide/Layers/Convolution-Layers.md#SpatialConvolution) layer, except that weights are unshared, that is, a different set of filters is applied at each different patch of the input.
 
     * `n_input_plane` The number of expected input planes in the image given into forward()
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: nInputPlane

**Comment by helenlly** [comment]
updated as Kai's comment

**Comment by yiheng** [comment]
@helenlly Please use cherry-pick to move this change to master

**Pull request merged**